lib/timed_controller.dart

```diff
@@ -56,20 +60,28 @@ class TimedController extends WidgetsBindingObserver {
       //Fetch network balances
       if (!_apiService.gettingNetworkBalances) {
         _apiService.getNetworkBalances(myIds).then((balances) {
-          debugPrint("Got balances for ${balances.length} IDs");
-          appStore.setAmounts(balances);
+          Map<String, int> changedIds = appStore.setAmounts(balances);
+          if (changedIds.isNotEmpty) {
+            _walletConnectService.triggerAmountChangedEvent(changedIds);
+          }
         }, onError: (e) {
           appStore
               .reportGlobalError(e.toString().replaceAll("Exception: ", ""));
-          //_globalSnackBar.show(e.toString().replaceAll("Exception: ", ""));
         });
       }
 
       //Fetch network assets
       if (!_apiService.gettingNetworkAssets) {
-        _apiService
-            .getCurrentAssets(myIds)
-            .then((assets) => appStore.setAssets(assets));
+        _apiService.getCurrentAssets(myIds).then((assets) {
```

Comment:
I don't understand the logic here. It seems to me the event is triggered any time the app is fetching the assets balance and not only when the new results are different than the previous time it was fetched, correct @Qubic-Hub ?

Reply:
@sallymoc  The setAmounts returns only the accounts whose value has changed from the previous time

Reply:
But if I'm not wrong, it's considering changes in all the accounts and not only the ones the user owns the private key.

Reply:
You are right. I have issued a fix for this